osquery/tables/system/windows/users.cpp

```diff
@@ -110,15 +104,38 @@ void processRoamingProfiles(const std::set<std::string>& processedSids,
     ret = LookupAccountSidW(
         nullptr, sid, accntName, &accntNameLen, domName, &domNameLen, &eUse);
     r["username"] = ret != 0 ? wstringToString(accntName) : "";
-    LocalFree(sid);
+  }
+  
+  results.push_back(r);
+}
+
+// Enumerate the users from the profiles key in the Registry, skipping any 
+// that are given in processedSids (i.e., that have already been processed)
+void processRoamingProfiles(const std::set<std::string>& processedSids,
+                            QueryData& results) {
+  QueryData regResults;
+  queryKey(kRegProfilePath, regResults);
 
-    results.push_back(r);
+  for (const auto& profile : regResults) {
+    Row r;
+    if (profile.at("type") != "subkey") {
+      continue;
+    }
+
+    auto sidString = profile.at("name");
+
+    // Skip this user if already processed
+    if (processedSids.find(sidString) == processedSids.end()) {
+      genUser(sidString, results);
+    }
   }
 }
 
+// Enumerate all local users
 void processLocalAccounts(std::set<std::string>& processedSids,
                           QueryData& results) {
-  unsigned long dwUserInfoLevel = 3;
+  unsigned long dwUserInfoLevel = 3; // 3 is valid for NetUserEnum(); 4 is not
```

Comment:
Specifically, I'm wondering if we can use a _lower_ amount, which might be faster. It seems silly to pull a bunch of data, then pull it again